bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/bridge/AutomowerBridgeHandler.java

```diff
@@ -0,0 +1,160 @@
+
+    private final HttpClient httpClient;
+
+    @NonNullByDefault
+    private static class AutomowerBridgePollingRunnable implements Runnable {
+        private final Logger logger = LoggerFactory.getLogger(AutomowerBridgePollingRunnable.class);
+
+        private final AutomowerBridgeHandler handler;
+        private final AutomowerBridge bridge;
+
+        private AutomowerBridgePollingRunnable(AutomowerBridgeHandler handler, AutomowerBridge bridge) {
+            this.handler = handler;
+            this.bridge = bridge;
+        }
+
+        @Override
+        public void run() {
+            MowerListResult automowers;
+            try {
+                automowers = bridge.getAutomowers();
+                handler.updateStatus(ThingStatus.ONLINE);
+                logger.info("Found {} automowers", automowers.getData().size());
+            } catch (AutomowerCommunicationException e) {
+                handler.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/comm-error-query-mowers-failed");
+                logger.warn("Unable to fetch automowers", e);
+            }
+        }
+    }
+
+    public AutomowerBridgeHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        if (bridge != null) {
+            stopAutomowerBridgePolling(bridge);
+            bridge = null;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing automower bridge handler.");
+        bridgeConfiguration = getConfigAs(AutomowerBridgeConfiguration.class);
+
+        final String appKey = bridgeConfiguration.getAppKey();
+        final String userName = bridgeConfiguration.getUserName();
+        final String password = bridgeConfiguration.getPassword();
+        if (appKey == null || appKey.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/conf-error-no-app-key");
+        } else if (userName == null || userName.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/conf-error-no-username");
+        } else if (password == null || password.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/conf-error-no-password");
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+            if (bridge == null) {
+                bridge = new AutomowerBridge(appKey, userName, password, httpClient, scheduler);
+                // bridge.setTimeout(5000);
+
+                startAutomowerBridgePolling(bridge);
+            }
+        }
+    }
+
+    private void startAutomowerBridgePolling(AutomowerBridge bridge) {
+        if (automowerBridgePollingJob == null || automowerBridgePollingJob.isCancelled()) {
```

Comment:
No harm in canceling a task that is already cancelled. Same for below.

Reply:
will do that